### ISSUE TITLE
0.8.1: pytest warnings

### ISSUE THREAD

**Issue by kloczek** [sarugaku/resolvelib#103]
I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

Here is pytest output:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-resolvelib-0.8.1-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-resolvelib-0.8.1-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/resolvelib-0.8.1
collected 48 items

tests/test_resolvers.py ....                                                                                                                                         [  8%]
tests/test_structs.py .......                                                                                                                                        [ 22%]
tests/functional/cocoapods/test_resolvers_cocoapods.py x........x...............                                                                                     [ 75%]
tests/functional/python/test_resolvers_python.py ....xx..                                                                                                            [ 91%]
tests/functional/swift-package-manager/test_resolvers_swift.py ....                                                                                                  [100%]

============================================================================= warnings summary =============================================================================
tests/functional/cocoapods/test_resolvers_cocoapods.py: 188 warnings
tests/functional/python/test_resolvers_python.py: 20 warnings
  /usr/lib/python3.8/site-packages/packaging/version.py:111: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
    warnings.warn(

tests/functional/python/test_resolvers_python.py: 27 warnings
  /usr/lib/python3.8/site-packages/packaging/specifiers.py:255: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
XFAIL tests/functional/cocoapods/test_resolvers_cocoapods.py::test_resolver[circular]
  circular dependencies works for us, no conflicts
XFAIL tests/functional/cocoapods/test_resolvers_cocoapods.py::test_resolver[fixed_circular]
  circular dependencies works for us, no backtracks
XFAIL tests/functional/python/test_resolvers_python.py::test_resolver[pyrex-1.9.8]
  Too many rounds (>500)
XFAIL tests/functional/python/test_resolvers_python.py::test_resolver[same-package-extras]
  State not cleaned up correctly
=============================================================== 44 passed, 4 xfailed, 235 warnings in 14.99s ===============================================================
```

**Comment by uranusjr** [comment]
Contributions would be much welcomed.

**Comment by frostming** [comment]
Closed by #112

**Comment by kloczek** [comment]
Cnnot test that PR because it cannot be applied cleanly on top of master.
